appnav/src/main/kotlin/io/element/android/appnav/root/RootNavStateFlowFactory.kt

```diff
@@ -73,6 +75,8 @@ class RootNavStateFlowFactory @Inject constructor(
                 matrixClientsHolder.remove(sessionId)
                 // Ensure image loader will be recreated with the new MatrixClient
                 imageLoaderHolder.remove(sessionId)
+                // Also remove cached value for SessionPreferencesStore
+                sessionPreferencesStoreFactory.remove(sessionId)
```

Comment:
This is actually the location of the fix for the blank screen after clear cache.